plugins/flatpak/gs-flatpak.c

```diff
@@ -1348,12 +1439,15 @@ gs_flatpak_add_updates (GsFlatpak *self, GsAppList *list,
 			g_warning ("failed to add flatpak: %s", error_local->message);
 			continue;
 		}
-		gs_app_set_state (app, AS_APP_STATE_UPDATABLE_LIVE);
-		gs_app_set_update_details (app, NULL);
-		gs_app_set_update_version (app, NULL);
-		gs_app_set_update_urgency (app, AS_URGENCY_KIND_UNKNOWN);
-		gs_app_set_size_download (app, 0);
-		gs_app_list_add (list, app);
+
+		main_app = get_real_app_for_update (self, app, cancellable, error);
```

Comment:
It looks like `get_real_app_for_update()` would always return a valid `GsApp*`, and never `NULL`, but that's not clear when reading the code here unless you check that function's implementation. However, I'm not entirely convinced that this function can never fail because the operation could get cancelled (see my other comment).

So, please add an assertion if this is actually guaranteed to never return NULL, or a NULL-check plus an early return otherwise.